lib/safe-pg-migrations/plugins/idem_potent_statements.rb

```diff
@@ -26,6 +26,10 @@ def remove_index(table_name, options = {})
       SafePgMigrations.say("/!\\ Index '#{index_name}' not found on table '#{table_name}'. Skipping statement.", true)
     end
 
+    def add_foreign_key(from_table, to_table, **options)
+      return super unless foreign_key_exists?(from_table, to_table)
```

Comment:
You might want to test it, but I think you also need to pass `**options` to `foreign_key_exists?`.
Not sure it works for statement like this right now:
```rb
add_foreign_key :posts, :users, column: :author_id
```

Reply:
I think we don't need it, because we can pass the options or the target table 
> foreign_key_exists?(from_table, options_or_to_table = {})

https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/foreign_key_exists%3F

Reply:
Not in Rails 6 it seems: https://apidock.com/rails/v6.0.0/ActiveRecord/ConnectionAdapters/SchemaStatements/foreign_key_exists%3F

Reply:
@rchoquet should be better now. I needed to remove `:validate` from options because otherwise it would return false even if the foreign key existed. Can you take a look ?